misc/Kconfig

```diff
@@ -44,9 +44,12 @@ endif
 config TEXT_SECTION_OFFSET
 	hex "TEXT section offset"
 	default 0
+	default 0x200 if BOOTLOADER_MCUBOOT
 	help
 	  This sets and offset before the text section and allows booting images
 	  with custom headers that are inserted at the beginning of the image.
+	  If the application is to be booted by MCUboot, a default suitable
+	  value is set in order to place the image's signature.
```

Comment:
Minor things:

- Runtimeco prefers the capitalization "MCUBoot" (note capital B)
- The signature is actually in the image trailer, not the header; TEXT_SECTION_OFFSET makes room for the header (see docs/design.md in the mcuboot repository for details).
- A smaller TEXT_SECTION_OFFSET default value would work to contain the image header, but it wouldn't meet vector table constraints on some "important" ARM targets (in particular, frdm_k64f), so the choice of 0x200 is architecture-dependent. I think it would be good to document this here.

How about this instead:

"If the application is built for chain-loading by MCUBoot using the BOOTLOADER_MCUBOOT option, a suitable default value for TEXT_SECTION_OFFSET is set with sufficient space to store the image header, which should also meet vector table alignment requirements on most ARM targets. Other targets may be able to use smaller values, or require larger ones."

Reply:
I like your proposal, I will include this instead of my original text